src/games/strategy/util/SHA256Crypt.java

```diff
@@ -0,0 +1,23 @@
+package games.strategy.util;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Replacement for MD5Crypt
+ * using  SHA256 instead
+ */
+public class SHA256Crypt {
+  /**
+   * Returns the SHA256-Hash of the given String
+   */
+  public static String crypt(String text) throws UnsupportedEncodingException, NoSuchAlgorithmException{
```

Comment:
Couple of problems with the API.

### static
This will make it hard to test since you can no longer control the output. Alternatively the test code will have to duplicate the library calls of the code being tested, which also undesirable.

Can be solved via dependency injection. Instead of:
```

public class foo { 
  public foo( ) {
     encrypted = SHA256Crypt.crypt( text );
   }
}
```
It would be:

```
public class foo { 
  public foo( ) {
     this(ClientContext.sha256Crypt();
  } 

 public foo( SHA256Crypt crypt) {
   encrypted = crypt.crypt( text );
   }
}
```

In the latter case, when we move to a proper DI framework, the empty `foo()` constructor could be removed. Until then we have a default API to use in prod code, and  we have an API that we can use for testing. The difference between the two is how the dependency is constructed, which is something the test case can deal with via mocks.

Also notice that in this case, if we create an interface, and use the interface, any future updates would only be inside of `ClientContext`, and not anywhere else in the code.

### Throwing API
Makes the code very hard to use from a client perspective: `throws UnsupportedEncodingException, NoSuchAlgorithmException`
If we get either of those, there is nothing much we can do. There is a startup check module where we should check for this once. Then we'll know at startup time this is broken rather than later when it is less convenient. Once that is done, some basic error handling is then all that is needed:
`ClientLogger.logError(e)`